include/openenclave/host.h

```diff
@@ -45,11 +45,23 @@ OE_EXTERNC_BEGIN
  */
 #define OE_ENCLAVE_FLAG_SIMULATE 0x00000002u
 
+/**
+ *  Flag passed into oe_create_enclave to enable the enclave with switchless
+ *  calls. A switchless call is a function invocation from the trusted world to
+ *  the untrusted world, or vice versa, without context switches and transition
+ *  of the execution mode. Typically the caller posts the job to a shared
+ *  memory buffer accessible by both the trusted world and the untrusted world,
+ *  and a workers in the other world takes the job and posts the result, which
+ *  is retrieved by the caller.
+ */
+#define OE_ENCLAVE_FLAG_CONTEXT_SWITCHLESS 0x00000004u
```

Comment:
Why is this flag necessary? There are already at least two mechanisms that do the same thing:

- A developer already has to explicitly decide whether to mark any methods in the EDL as `transition_using_threads` and it seems weirdly redundant to ask them to specify an additional flag on enclave creation if they've already expressed this intention. Particularly since we require that they call the EDL generated wrapper for enclave creation, it seems like edger8r should be able to already tell if switchless manager is necessary or not.

- There's an added option to pass the number of host worker threads as part of `config` to `oe_create_enclave()`. Why isn't this just equivalent to passing 0 to that function if you explicitly want to specify that no switchless calls should be used (and are not using the edger8r wrapper)?

Reply:
Completely agree with Simon, this flag should not be needed.

Reply:
We need this flag so in later call to `oe_create_enclave`, we know the `void*` parameter is actually for configuring switchless.  

If we make the switchless setting a TLV type, then I agree we don't need this flag.

Reply:
The TLV type of configs is implemented. Removing the flag.